File: 1972-rotating-the-box/rotating-the-box.java
class Solution{
    public char[][] rotateTheBox(char[][] b){
        int n=b.length;
        int m=b[0].length;

        char[][] dp=new char[m][n];
        for(char[] ch: dp){
            Arrays.fill(ch, '.');
        }
        for(int r=0; r<n; r++){
            int i=m-1;
            for(int c=m-1; c>=0; c--){
                if(b[r][c] == '#'){
                    dp[i][n-r-1]='#';
                    i--;
                }
                else if(b[r][c] == '*'){
                    dp[c][n-r-1]='*';
                    i=c-1;
                }
            }
        }

        return dp;
    }
}

File: Medium
Medium/1972-rotating-the-box/rotating-the-box.py
class Solution:
    def rotateTheBox(self, b: List[List[str]]) -> List[List[str]]:
        n=len(b)
        m=len(b[0])

        dp=[['.' for _ in range(n)] for _ in range(m)]
        for r in range(n):
            i=m-1
            for c in range(m-1, -1, -1):
                if b[r][c] == '#':
                    dp[i][n-r-1]='#'
                    i-=1
                elif b[r][c] == '*':
                    dp[c][n-r-1]='*'
                    i=c-1

        return dp
        